service/src/main/java/edu/tamu/catalog/service/FolioCatalogService.java

```diff
@@ -276,4 +286,67 @@ private void buildCoreItem(String instanceId, String barcode, NodeList nodes, Ma
         catalogItems.put(barcode, itemData);
     }
 
+    @Override
+    public FeesFines getFeesFines(String uin) {
+        String path = "patron/account/";
+        String additional = "&includeLoans=false&includeCharges=true&includeHolds=false";
+        String url = String.format("%s%s%s?apikey={apikey}%s", getAPIBase(), path, uin, additional);
+        String apiKey = getAuthentication().get(CatalogServiceFactory.FIELD_APIKEY);
+
+        logger.debug(String.format("Asking for fines from: %s", url));
+
+        JsonNode node = restTemplate.getForObject(url, JsonNode.class, apiKey);
+
+        Double total = null;
+        if (node.has("totalCharges") && node.get("totalCharges").has("amount")) {
+            total = node.get("totalCharges").get("amount").asDouble();
+        }
+
+        List<FeeFine> list = new ArrayList<>();
+
+        if (node.has("charges")) {
```

Comment:
It would be nice to use constants over string literals. Especially the ones that are used multiple times.

Reply:
As it will likely get busy, I recommend a FolioConstants class to hold these.

Reply:
I think we might be able to leave this concern to after we have the existing cards completed and can do some structural cleanups.

Reply:
Yeah, it works as is. Will be more chances to come up with something elegant to parse JsonNode.